bittide-instances/bin/TerminalApi.hs

```diff
@@ -0,0 +1,26 @@
+-- SPDX-FileCopyrightText: 2023 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+import Prelude
+
+import Control.Exception
+import Paths_bittide_instances
+import System.Directory
+import System.Environment
+import Data.List(intersperse)
+
+main :: IO ()
+main = do
+  args <- getArgs
+  case args of
+    "getDataFileName":[file] -> do
+      fPath <- getDataFileName file
+      fileExists <- doesFileExist fPath
+      if fileExists
+        then putStrLn fPath
+        else throwIO
+          (userError $ "File " <> show file <> " does not exist in bittide-instances.")
+    invalidInput ->
+      throwIO (userError $ "Invalid input: " <> (concat $ intersperse " " invalidInput))
+  pure ()
```

Comment:
```suggestion
main :: IO ()
main = do
  args <- getArgs
  case args of
    [file] -> do
      fPath <- getDataFileName file
      fileExists <- doesFileExist fPath
      if fileExists 
      then putStrLn fPath
      else throwIO (userError $ unwords 
             ["File", show file, "does not exist in bittide-instances."])
    [] -> throwIO (userError "Expected one file argument, got none")
    _  -> throwIO (userError $ "Expected one file argument, got: " <> unwords args)
  pure ()
```

Let's keep it simple. I can't really imagine what else we'd wanna do here - I suggest renaming the binary to `get-data-file-name`. That would make the invocation:

```
cabal run -- bittide-instances:get-data-file-name "foo/bar.txt"
```